data-loader/src/v2/ConsumableDataParser.ts

```diff
@@ -0,0 +1,178 @@
+import * as fs from 'fs';
+import { parseCsv } from './parseCsv';
+
+import {
+  IEntityGraph,
+  IEntityNode,
+  EEntityType,
+  ParserOptions,
+  IConsumablesCSVData,
+} from './types';
+
+export class ConsumableDataParser {
+  private readonly CONSUMABLE_CODE = '77fcbb00';
+  private readonly CONSUMABLE_NAME = 'Consumable';
+
+  public readonly path: fs.PathLike;
+  public readonly options?: ParserOptions;
+
+  private data: IConsumablesCSVData;
+
+  private isParsed: boolean;
+  private isBuilt: boolean;
+
+  constructor(path: fs.PathLike, options?: ParserOptions) {
+    this.path = path;
+    this.options = options;
+
+    this.isParsed = false;
+    this.isBuilt = false;
+
+    this.data = [];
+  }
+
+  public async parseData(): Promise<IConsumablesCSVData> {
+    if (this.isParsed) return this.data;
+
+    this.data = await parseCsv(this.path, this.options);
+
+    this.isParsed = true;
+    return this.data;
+  }
+
+  public buildConsumableNode(graph: IEntityGraph): IEntityNode {
+    if (this.isBuilt) return graph[this.CONSUMABLE_CODE];
+
+    // Initialise adjacency list for storing graph.
+    graph[this.CONSUMABLE_CODE] = {
+      code: this.CONSUMABLE_CODE,
+      name: this.CONSUMABLE_NAME,
+      description: this.CONSUMABLE_NAME,
+      type: EEntityType.Category,
+      children: [],
+      properties: [],
+    };
+
+    try {
+      // Initialise duplicate codes.
+      const duplicates: string[] = [];
+
+      // Parse entity graph.
+      this.data.forEach(row => {
+        const productDefinition = [
+          // TODO: leaving this as Product for now... do we want that??
```

Comment:
Yes I think so, as I'm using Product to find the top level of the tree to display when navigating to a lower level code